src/main/java/net/helix/pendulum/service/milestone/impl/LatestSolidMilestoneTrackerImpl.java

```diff
@@ -91,6 +91,15 @@
      */
     private int repairBackoffCounter = 0;
 
+    private static final LatestSolidMilestoneTrackerImpl INSTANCE = new LatestSolidMilestoneTrackerImpl();
+
+    private LatestSolidMilestoneTrackerImpl() {
+    }
+
+    public static LatestSolidMilestoneTrackerImpl getInstance() {
```

Comment:
can we return interfaces, not classes here?

Reply:
Yes, it's possible, but if exact implementation returns the corresponding instance, it's ok. If we like to use interface, the vars has to be declared as interfaces.

Reply:
I can update the corresponding vars declaration. Ok?